sympy/stats/frv_types.py

```diff
@@ -118,12 +121,10 @@ def DiscreteUniform(name, items):
 class DieDistribution(SingleFiniteDistribution):
     _argnames = ('sides',)
 
-    def __new__(cls, sides):
-        sides_sym = sympify(sides)
-        if fuzzy_not(fuzzy_and((sides_sym.is_integer, sides_sym.is_positive))):
-            raise ValueError("'sides' must be a positive integer.")
-        else:
-            return super(DieDistribution, cls).__new__(cls, sides)
+    @staticmethod
+    def check(sides):
+        _value_check(sides.is_positive and sides.is_integer,
```

Comment:
This needs the fuzzy logic or it will evaluate to False if one is unknown. The idiom has been to raise an error only if the condition is False, not if it is unevaluated.

Reply:
i will update this asap.

Reply:
`_whenFalse(fuzzy_and([sides.is_positive, sides.is_integer]), msg)`